shell/platform/common/cpp/text_input_model.cc

```diff
@@ -155,22 +158,20 @@ bool TextInputModel::DeleteSurrounding(int offset_from_cursor, int count) {
 }
 
 bool TextInputModel::MoveCursorToBeginning() {
-  if (selection_base_ == text_.begin() && selection_extent_ == text_.begin())
+  if (selection_base_ == 0 && selection_extent_ == 0)
     return false;
 
-  selection_base_ = text_.begin();
-  selection_extent_ = text_.begin();
-
+  selection_base_ = 0;
+  selection_extent_ = 0;
   return true;
 }
 
 bool TextInputModel::MoveCursorToEnd() {
-  if (selection_base_ == text_.end() && selection_extent_ == text_.end())
+  if (selection_base_ == text_.length() && selection_extent_ == text_.length())
```

Comment:
Nit: maybe extract `text_.length()` to an ivar so we're not calling it four times.

Reply:
Done here and in SetSelection.